### PR TITLE
Update the release workflows

### PR DESCRIPTION
**Pull request by collins-w**
- Refactor the release workflows following the publishing flow

### CODE REVIEW

File: .github/workflows/rc-push.yml [hunk 1]
```diff
@@ -61,8 +62,8 @@ jobs:
       # Give the default GITHUB_TOKEN write permission to commit and push the
       # added or changed files to the repository.
       contents: write
+      pull-requests: write
     needs: verify-packages
-    if: needs.verify-packages.outputs.changed_packages > 0
```
Comment (tirumerla) [inline .github/workflows/rc-push.yml:65]: You should keep this, this was removed to test the functionality

File: .github/workflows/stable-publish.yml [hunk 2]
```diff
@@ -1,9 +1,8 @@
 name: Publish stable
 
 on:
-  push:
-    branches:
-      - master
+  workflow_dispatch: {}
+  workflow_call: {}
```
Comment (tirumerla) [inline .github/workflows/stable-publish.yml:5]: keep on push branches here and remove workflow_dispatch, it is only for stable-push , this should detect the author and if the commit message has rc or not

File: .github/workflows/stable-push.yml [hunk 3]
```diff
@@ -90,3 +89,12 @@ jobs:
           branch: ${{ github.head_ref || github.ref_name }}
         env:
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (tirumerla) [inline .github/workflows/stable-push.yml:91]: Change this to use the pull request github action

File: .github/workflows/stable-push.yml [hunk 4]
```diff
@@ -90,3 +89,12 @@ jobs:
...
+
+  publish-stable-version:
+    name: Stable Publish
+    needs: tag
+    permissions:
+      contents: write
+    needs: verify-packages
+    if: needs.verify-packages.outputs.changed_packages > 0
+    uses: ./.github/workflows/stable-publish.yml
```
Comment (tirumerla) [inline .github/workflows/stable-push.yml:100]: We don't need this, please update similar to what we have in `rc-push.yaml`

**Review by tirumerla — CHANGES_REQUESTED** [review]
Added comments, need changes.

**Review by tirumerla — CHANGES_REQUESTED** [review]

File: .github/workflows/stable-push.yml [hunk 5]
```diff
@@ -82,11 +81,28 @@ jobs:
...
-      - uses: planetscale/ghcommit-action@6ca6b06e9d2de1296143831c53e1d601831ba2a2
+      # # Commit all changed files back to the repository
+      # - uses: planetscale/ghcommit-action@6ca6b06e9d2de1296143831c53e1d601831ba2a2
+      #   with:
+      #     commit_message: "🤖 Update version to ${{ steps.update_version.outputs.stable_version }}"
+      #     repo: ${{ github.repository }}
+      #     branch: ${{ github.head_ref || github.ref_name }}
+      #   env:
+      #     GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
```
[the remark this answers is not in the document]
Comment (collins-w) [reply .github/workflows/stable-push.yml]: This was not intentional.

**Review by tirumerla — APPROVED** [review]
lgtm

**Pull request merged**